### ISSUE TITLE
Синхронизация 

### ISSUE THREAD

**Issue by perevozchikov85** [gee12/MyTetroid#9]
Вроде в настройках появилась синхронизация, не понял как она работает. У меня стоит МГИТ. Включил срхоонизацию при запуске, что прописывать в настройаах сирхоонизации?

**Comment by gee12** [comment]
Пропустил уведомление об этом issue, поэтому отвечаю только сейчас.

Есть моя неофициальная версия MGit с функционалом синхронизации из MyTetroid (вот [apk](https://yadi.sk/d/F7aNnR-7Ta495w)).
В офф.репозиторий MGit я отправил [pullRequest](https://github.com/maks/MGit/pull/488), но разработчик выкатил довольно много правок, на которые, честно сказать, лень отвлекаться, пока не закончил с реализацией редактирования записей.
И да, вызов обновления хранилища (pull) из MGit работает НЕ ВСЕГДА, по крайней мере у меня так.

**Comment by kb-1000** [comment]
I would recommend you to not call into another app (MGit in this case) for synchronization purposes but rather use the underlying library (JGit) directly. That way, you have much more control about what it does, too.

**Comment by perevozchikov85** [comment]
не получается засинхронизировать базу все время ошибка 
![Screenshot_20200417-225642365](https://user-images.githubusercontent.com/43738298/79604387-0eb54d00-80ff-11ea-8ad6-73b6284df44c.jpg)

**Comment by perevozchikov85** [comment]
А нет вроде всё норм.
Какую команду прописать в настройках приложения чтобы и отправлялись данные и получались данные с репозитория?

`git pull; git push`

**Comment by gee12** [comment]
> Какую команду прописать в настройках приложения чтобы и отправлялись данные и получались данные с репозитория?

В MGit была добавлена лишь обработка команды `pull`. Но такая связка себя не оправдала (работает нестабильно), поэтому я не добавлял обработку `push` и подумываю над другим способом синхронизации хранилища - git-репозитория.

**Comment by perevozchikov85** [comment]
Хорошо, значит ждём независимой синхронизации :)

**Comment by sdfq** [comment]
Кстати нашел более удобный способ синхронизации правда ручной с помощью termux, необходима настройка. Создал ключи ssh  подключил свою удаленную репу к приложению. Установил termux widget. Создал в папке .shortcuts sh  скрипт такого вида:

![Screenshot_20200802-191130718](https://user-images.githubusercontent.com/67936295/89126121-4a302780-d4f4-11ea-857b-9c242054e81b.jpg)

Подключил этот скрип к termux widget, теперь на рабочем столе телефона у меня виджет с названием mytetrasync, при запуске которого происходит синхоонизация.

**Comment by gee12** [comment]
Да, можно использовать этот вариант (на Android >= 7). Я недавно обновил
статью:  https://gee12.space/sinhronizacija-mytetroid-na-android-git/, но
здесь не отписался.

вс, 2 авг. 2020 г. в 18:15, sdfq <notifications@github.com>:

**Comment by Shura0** [comment]
Ещё один git клиент https://github.com/catpuppyapp/PuppyGit
Он пока не умеет запускать синхронизацию по выполнению внешней команды (хотя автор вроде [не против](https://github.com/catpuppyapp/PuppyGit/issues/80) организовать такой вариант), но программа может выполнять синхронизацию при запуске и при завершении выбранного приложения (mytetroid, например). Но есть небольшая проблема. Если синхронизация выполняется при запуске, то mytetroid успевает считать файлы до того, как puppygit выполнит синхронизацию
